src/main/java/com/amazon/dlic/auth/ldap/util/LdapHelper.java

```diff
@@ -34,37 +34,46 @@
 import org.ldaptive.referral.SearchReferralHandler;
 
 import org.opensearch.SpecialPermission;
+import org.opensearch.common.settings.Settings;
 
 public class LdapHelper {
 
     private static SearchFilter ALL = new SearchFilter("(objectClass=*)");
+
     @SuppressWarnings("removal")
     public static List<LdapEntry> search(final Connection conn, final String unescapedDn, SearchFilter filter,
-            final SearchScope searchScope, final String[] returnAttributes) throws LdapException {
+            final SearchScope searchScope, final String[] returnAttributes, Settings settings) throws LdapException {
 
         final SecurityManager sm = System.getSecurityManager();
 
         if (sm != null) {
             sm.checkPermission(new SpecialPermission());
         }
-
         try {
             final String baseDn = escapeDn(unescapedDn);
             return AccessController.doPrivileged(new PrivilegedExceptionAction<List<LdapEntry>>() {
                 @Override
                 public List<LdapEntry> run() throws Exception {
                     final List<LdapEntry> entries = new ArrayList<>();
                     final SearchRequest request = new SearchRequest(baseDn, filter);
```

Comment:
One of the other requests in the source ticket was if there was any failures when attempting to resolve the referrals a partial response could still be returned [see comment](https://github.com/opensearch-project/security/issues/195#issuecomment-717058416)  I'm not sure I agree with this behavior, I think its easier not to swallow failures - did you consider that scenario, what do you think?

Reply:
I do not think partial responses are the solution since it seems like it would cause further issues for any code expecting full responses. This would likely just move whatever error is resolved by adding the `shouldFollowReferrals` field further down the line. 

I could be wrong however.

Reply:
Behavior has been updated to elicit an Ldap response of `dn="cn=couldNotFollowReferrals"` when `shouldFollowReferrals` is `false` and following is attempted.